### PR TITLE
Fix using directives not working with the shebang line in `.scala` files

### LINKED ISSUE #1638: shebang stopped working after 0.1.18
**Version(s)**

~~~ text
$ scala-cli --version
Scala CLI version: 0.1.18
Scala version (default): 3.2.1
~~~

**Describe the bug**

Scala-CLI ignores the “using directives” of my executable script that uses the shebang command.

**To Reproduce**

Consider a file `Test.scala`:

~~~ text
$ ls -l          
total 4
-rwxrwxr-x 1 julien julien 288 Dec  4 15:56 Test.scala
~~~

It contains a regular Scala-CLI script and uses the [shebang command](https://scala-cli.virtuslab.org/docs/reference/commands/#shebang):

~~~ text
$ cat Test.scala 
#!/usr/bin/env -S scala-cli shebang
//> using scala "2.13.8"
//> using repository "https://artifacts.unidata.ucar.edu/repository/unidata-all"
//> using lib "edu.ucar:grib:5.3.1"

import ucar.nc2.dt.grid.{GeoGrid, GridDataset}

object Main {
  def main(args: Array[String]): Unit = ()
}
~~~

Note that the script uses Scala 2.13.8, and one library dependency.

Trying to run it (on Ubuntu):

~~~ text
$ ./Test.scala 
Compiling project (Scala 3.2.1, JVM)
[error] ./Test.scala:6:8: Not found: ucar
[error] import ucar.nc2.dt.grid.{GeoGrid, GridDataset}
[error]        ^^^^
Error compiling project (Scala 3.2.1, JVM)
Compilation failed
~~~

The import is not resolved, and the Scala version is wrong.

Note that the regression seems to be pretty recent as it was working not a long time ago.

**Expected behaviour**

Scala-CLI does take into account the using directives of the file.

**Workaround**

- Remove the shebang on the first line
- Run the script with `scala-cli run Test.scala -- ...`

### PR DESCRIPTION
**Pull request by Gedochao**
Fixes #1638

### CODE REVIEW

File: modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala [hunk 1]
```diff
@@ -1024,4 +1024,29 @@ abstract class RunTestDefinitions(val scalaVersionOpt: Option[String])
         expect(output.contains(exceptionMsg))
       }
     }
+
+  test("should respect using directives in a .scala file with the shebang line") {
```
Comment (lwronski) [inline modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala]: In this case, we should add unit tests. It should be enough to add a unit test and verify that the using directives are parsed correctly.

**Review by lwronski — CHANGES_REQUESTED** [review]

File: modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala (hunk 1 above)
Comment (Gedochao) [reply modules/integration/src/test/scala/scala/cli/integration/RunTestDefinitions.scala]: should be fixed

File: modules/build/src/test/scala/scala/build/tests/ScalaPreprocessorTests.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      val scalaFile = SourceScalaFile(root, os.sub / "sample.sc")
+      val Some(Right(result)) = ScalaPreprocessor.preprocess(
+        scalaFile,
+        logger = TestLogger(),
+        allowRestrictedFeatures = false
+      )
+      expect(result.nonEmpty)
+      val Some(directivesPositions) = result.head.directivesPositions
+      expect(directivesPositions.specialCommentDirectives.startPos == 0 -> 0)
+      println(directivesPositions.specialCommentDirectives.endPos)
```
Comment (lwronski) [inline modules/build/src/test/scala/scala/build/tests/ScalaPreprocessorTests.scala]:
```suggestion
```

**Review by lwronski — APPROVED** [review]
LGTM

**Pull request merged**